src/textual/drivers/linux_driver.py

```diff
@@ -102,19 +102,29 @@ def _get_terminal_size(self) -> tuple[int, int]:
         Returns:
             The size of the terminal as a tuple of (WIDTH, HEIGHT).
         """
-        width: int | None = 80
-        height: int | None = 25
-        import shutil
+        try:
+            width = int(os.environ['COLUMNS'])
+        except (KeyError, ValueError):
+            width = 0
 
         try:
-            width, height = shutil.get_terminal_size()
-        except (AttributeError, ValueError, OSError):
+            height = int(os.environ['LINES'])
+        except (KeyError, ValueError):
+            height = 0
+
+        if width <= 0 or height <= 0:
             try:
-                width, height = shutil.get_terminal_size()
+                size = os.get_terminal_size(self._file.fileno())
             except (AttributeError, ValueError, OSError):
-                pass
-        width = width or 80
-        height = height or 25
+                try:
+                    size = os.get_terminal_size(self._file.fileno())
+                except (AttributeError, ValueError, OSError):
+                    pass
+
+            if columns <= 0: 
+                width = size.columns or 80
+            if lines <= 0:
+                height = size.lines or 25
```

Comment:
`columns` and `lines`? I think you meant `width` and `height`.

Reply:
Additionally, if we hit `pass` at runtime, size may be unset, and trying to reach `size.columns` or `size.lines` would yield a NameError.

Reply:
I went digging inside Rich and Textual history: that double try-except we were trying to preserve here was initially meant to try `os.get_terminal_size(sys.stdin.fileno())` then `os.get_terminal_size(sys.stdout.fileno())`... but that no longer makes sense here:
- stdout may already be a pipe
- stdin is currently used to read terminal events but it could become a pipe in the future too
- stderr (i.e. `self._file`) is where Textual renders applications

Otherly put: we can **discard that double try-except**, which should make things easier.